atc/api/versions_test.go

```diff
@@ -86,12 +85,128 @@ var _ = Describe("Versions API", func() {
 				BeforeEach(func() {
 					fakePipeline.PublicReturns(true)
 					fakePipeline.ResourceReturns(fakeResource, true, nil)
-					fakeResource.VersionsReturns([]atc.ResourceVersion{}, db.Pagination{}, true, nil)
+
+					returnedVersions := []atc.ResourceVersion{
+						{
+							ID:      4,
+							Enabled: true,
+							Version: atc.Version{
+								"some": "version",
+							},
+							Metadata: []atc.MetadataField{
+								{
+									Name:  "some",
+									Value: "metadata",
+								},
+							},
+						},
+						{
+							ID:      2,
+							Enabled: false,
+							Version: atc.Version{
+								"some": "version",
+							},
+							Metadata: []atc.MetadataField{
+								{
+									Name:  "some",
+									Value: "metadata",
+								},
+							},
+						},
+					}
+
+					fakeResource.VersionsReturns(returnedVersions, db.Pagination{}, true, nil)
 				})
 
 				It("returns 200 OK", func() {
 					Expect(response.StatusCode).To(Equal(http.StatusOK))
 				})
+
+				It("returns content type application/json", func() {
+					Expect(response.Header.Get("Content-type")).To(Equal("application/json"))
+				})
+
+				Context("when resource is public", func() {
+					BeforeEach(func() {
+						fakeResource.PublicReturns(true)
+					})
+
+					It("returns the json", func() {
+						body, err := ioutil.ReadAll(response.Body)
+						Expect(err).NotTo(HaveOccurred())
+
+						Expect(body).To(MatchJSON(`[
+					{
+						"id": 4,
+						"enabled": true,
+						"version": {"some":"version"},
+						"metadata": [
+							{
+								"name":"some",
+								"value":"metadata"
+							}
+						]
+					},
+					{
+						"id":2,
+						"enabled": false,
+						"version": {"some":"version"},
+						"metadata": [
+							{
+								"name":"some",
+								"value":"metadata"
+							}
+						]
+					}
+				]`))
+					})
+				})
+
+				Context("when resource is not public", func() {
+					Context("when the user is not authenticated", func() {
+						It("returns the json without version metadata", func() {
+							body, err := ioutil.ReadAll(response.Body)
+							Expect(err).NotTo(HaveOccurred())
+
+							Expect(body).To(MatchJSON(`[
+								{
+									"id": 4,
+									"enabled": true,
+									"version": {"some":"version"}
+								},
+								{
+									"id":2,
+									"enabled": false,
+									"version": {"some":"version"}
+								}
+							]`))
+						})
+					})
+
+					Context("when the user is authenticated", func() {
```

Comment:
do we need a test for `IsAuthorized` now? these both just show that it doesn't include version metadata, but we do want to include it if they're authorized

Reply:
These two tests are under the context "when not authorized". The "authorized" case is in a different context:

https://github.com/concourse/concourse/blob/7f05064caefc52487f95b30b695bb9c37ccfa601/atc/api/versions_test.go#L213

Reply:
ahh, makes sense - thanks!